staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -357,6 +360,18 @@ func cleanANSIEscapeCodes(s string) string {
 	return builder.String()
 }
 
+func validateInteractiveMode(mode clientcmdapi.ExecInteractiveMode) error {
+	switch mode {
+	case "":
+		// Empty mode is fine, for now.
+		return nil
```

Comment:
I think we can add defaulting to the beta API (but not v1) that would set this as `IfAvailableExecInteractiveMode` and have the rest of the code assume it is always set.  So the empty string case here would return a validation error similar to `field.Required`.

Reply:
Since this `InteractiveMode` field would be on the v1 _clientcmd_ API, would we write a defaulting func that looks at the `ExecConfig.APIVersion` to default the `ExecConfig.ExecInteractiveMode`? I tried that here: 54dd5407693621307d2dc61a9bcc2c3e832d472e.